src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -243,6 +244,22 @@ public function dump(array $options = [])
 EOF;
             $files = [];
 
+            if ($preload) {
+                $lineage = [];
+
+                $this->collectLineage($baseClassWithNamespace, $lineage, true);
+                $this->collectLineage(EnvPlaceholderParameterBag::class, $lineage);
+
+                foreach ($lineage as $file) {
+                    if (!isset($this->inlinedRequires[$file])) {
+                        $this->inlinedRequires[$file] = true;
+                        $preload .= sprintf("\n            include_once %s;", preg_replace('#^\\$this->targetDirs\[(\d++)\]#', '\dirname(__DIR__, 1 + $1)', $file));
```

Comment:
@nicolas-grekas should we drop this `$this->targetDirs` and always rely on `dirname` being resolved at compile time, even for normal files ?

Reply:
We could for some but not for all because the cache directory is a variable, to allow sandboxed cache clears (see arguments of the dumped constructor). So not sure it's worth the effort.